### PR TITLE
improve the logback config to hopefully work better in datadog

### PR DESCRIPTION
**Pull request by bystam**
# Jira Issue: [HVG-1050] 

## What?
- 

## Why?
- 

## Optional checklist
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

[HVG-1050]: https://hedvig.atlassian.net/browse/HVG-1050

### CODE REVIEW

File: src/main/resources/logback-spring.xml [hunk 1]
```diff
@@ -25,71 +24,52 @@
...
-                    <timestamp>
-                        <timeZone>UTC</timeZone>
-                    </timestamp>
-                    <!-- Assign logger fields to JSON object -->
+    <!-- Appender to log to file in a JSON format, one JSON object per line -->
+    <appender name="JSON_FILE" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder class="net.logstash.logback.encoder.LoggingEventCompositeJsonEncoder">
+            <providers>
+                <arguments/>
+                <mdc/>
```
Comment (FredrikAreschoug) [inline src/main/resources/logback-spring.xml:38]: :cool:

**Review by FredrikAreschoug — APPROVED** [review]
:ice_cream:

File: src/test/kotlin/com/hedvig/underwriter/service/UnderwriterImplValidateAndCompleteQuoteTest.kt [hunk 2]
```diff
@@ -117,5 +117,5 @@ private object SuccessShouldNotSkipAfter : BaseGuideline<QuoteData> {
     override val skipAfter: Boolean
         get() = false
 
-    override val validate : (QuoteData)-> Boolean = { false }
+    override val validate: (QuoteData) -> Boolean = { false }
```
Comment (bystam) [inline src/test/kotlin/com/hedvig/underwriter/service/UnderwriterImplValidateAndCompleteQuoteTest.kt:120]: auto format 🤷

File: src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -25,71 +24,53 @@
...
-                            }
-                        </pattern>
+                        {
+                        "severity": "%level",
+                        "service": "${springAppName:-}",
+                        "thread": "%thread",
+                        "logger": "%logger",
+                        "message": "%message"
+                        }
                     </pattern>
```
Comment (FredrikAreschoug) [inline src/main/resources/logback-spring.xml:53]: Just to be sure we want it to be:
```
<pattern>
   <pattern>
     ....
   </pattern>
</pattern>
```

**Review by FredrikAreschoug — APPROVED** [review]
Added a question if the awnser is yes then :ship: it

File: src/main/resources/logback-spring.xml (hunk 3 above)
Comment (bystam) [reply src/main/resources/logback-spring.xml:53]: It was like that before, and a quick look at the internet has other people doing it like that: https://dev.to/miranda/sending-formatted-logs-to-elk-2lb0

I think it is related to what the corresponding POJO representation of the config for `LoggingEventCompositeJsonEncoder` looks like, and I guess this one has a nested pattern structure. Some other encoders I saw online didn't have them nested.

Comment (bystam) [reply src/main/resources/logback-spring.xml:53]: So in short, yes!

**Pull request merged**